wakuv2/nwaku.go

```diff
@@ -3,6 +3,2350 @@
+	if err != nil {
+		return "", err
+	}
+
+	w.keyMu.Lock()
+	defer w.keyMu.Unlock()
+
+	if w.privateKeys[id] != nil {
+		return "", fmt.Errorf("failed to generate unique ID")
+	}
+	w.privateKeys[id] = key
+	return id, nil
+}
+
+// DeleteKeyPair deletes the specified key if it exists.
+func (w *Waku) DeleteKeyPair(key string) bool {
+	deterministicID, err := toDeterministicID(key, common.KeyIDSize)
+	if err != nil {
+		return false
+	}
+
+	w.keyMu.Lock()
+	defer w.keyMu.Unlock()
+
+	if w.privateKeys[deterministicID] != nil {
+		delete(w.privateKeys, deterministicID)
+		return true
+	}
+	return false
+}
+
+// AddKeyPair imports a asymmetric private key and returns it identifier.
+func (w *Waku) AddKeyPair(key *ecdsa.PrivateKey) (string, error) {
+	id, err := makeDeterministicID(hexutil.Encode(crypto.FromECDSAPub(&key.PublicKey)), common.KeyIDSize)
+	if err != nil {
+		return "", err
+	}
+	if w.HasKeyPair(id) {
+		return id, nil // no need to re-inject
+	}
+
+	w.keyMu.Lock()
+	w.privateKeys[id] = key
+	w.keyMu.Unlock()
+
+	return id, nil
+}
+
+// SelectKeyPair adds cryptographic identity, and makes sure
+// that it is the only private key known to the node.
+func (w *Waku) SelectKeyPair(key *ecdsa.PrivateKey) error {
+	id, err := makeDeterministicID(hexutil.Encode(crypto.FromECDSAPub(&key.PublicKey)), common.KeyIDSize)
+	if err != nil {
+		return err
+	}
+
+	w.keyMu.Lock()
+	defer w.keyMu.Unlock()
+
+	w.privateKeys = make(map[string]*ecdsa.PrivateKey) // reset key store
+	w.privateKeys[id] = key
+
+	return nil
+}
+
+// DeleteKeyPairs removes all cryptographic identities known to the node
+func (w *Waku) DeleteKeyPairs() error {
+	w.keyMu.Lock()
+	defer w.keyMu.Unlock()
+
+	w.privateKeys = make(map[string]*ecdsa.PrivateKey)
+
+	return nil
+}
+
+// HasKeyPair checks if the waku node is configured with the private key
+// of the specified public pair.
+func (w *Waku) HasKeyPair(id string) bool {
+	deterministicID, err := toDeterministicID(id, common.KeyIDSize)
+	if err != nil {
+		return false
+	}
+
+	w.keyMu.RLock()
+	defer w.keyMu.RUnlock()
+	return w.privateKeys[deterministicID] != nil
+}
+
+// GetPrivateKey retrieves the private key of the specified identity.
+func (w *Waku) GetPrivateKey(id string) (*ecdsa.PrivateKey, error) {
+	deterministicID, err := toDeterministicID(id, common.KeyIDSize)
+	if err != nil {
+		return nil, err
+	}
+
+	w.keyMu.RLock()
+	defer w.keyMu.RUnlock()
+	key := w.privateKeys[deterministicID]
+	if key == nil {
+		return nil, fmt.Errorf("invalid id")
+	}
+	return key, nil
+}
+
+// GenerateSymKey generates a random symmetric key and stores it under id,
+// which is then returned. Will be used in the future for session key exchange.
+func (w *Waku) GenerateSymKey() (string, error) {
+	key, err := common.GenerateSecureRandomData(common.AESKeyLength)
+	if err != nil {
+		return "", err
+	} else if !common.ValidateDataIntegrity(key, common.AESKeyLength) {
```

Comment:
no need for `else` after `return`
```suggestion
	}
	if !common.ValidateDataIntegrity(key, common.AESKeyLength) {
```

Reply:
yap, same 😶 

https://github.com/status-im/status-go/blob/d12f92d9dbbf4880477a1d3b3a5d59ceb6ebb535/wakuv2/gowaku.go#L937